Productionize background tx signature validation

bboston7 opened the pull request:
Partially addresses stellar/stellar-core-internal#377

This change:
* Centralizes signature checking logic between background validation and the `checkValid` flow.
* Checks operation signatures in the background
* Checks inner transaction signatures for fee bumps in the background
* Adds additional testing around edge cases

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

bboston7 commented on src/transactions/OperationFrame.cpp:
```diff
@@ -215,7 +215,7 @@ OperationFrame::isOpSupported(LedgerHeader const&) const
 
 bool
 OperationFrame::checkSignature(SignatureChecker& signatureChecker,
-                               LedgerSnapshot const& ls, OperationResult& res,
+                               LedgerSnapshot const& ls, OperationResult* res,
                                bool forApply) const
```
I think this is a false positive from the copilot instructions that push copilot to double check that any transaction changes are also reflected in fee bumps. There is no fee bump specific operation frame however, so there is no "fee bump transaction support for operation signature checking" to add. This PR does cover signatures on fee bump transactions themselves.

SirTyson reviewed (changes requested):
Overall looks good to me, some helpful cleanup! A few nit pics, then one potential race condition. I don't think it would actually cause a bug given the way the cache is structured. The bigger issue is that getting/updating a BucketList snapshot is relatively expensive and should be minimized when possible.

bboston7 commented on docs/metrics.md:
```diff
@@ -56,6 +56,9 @@ bucketlistDB.cache.bytes                  | counter   | estimated size in bytes
 crypto.verify.hit                         | meter     | number of signature cache hits
 crypto.verify.miss                        | meter     | number of signature cache misses
 crypto.verify.total                       | meter     | sum of both hits and misses
+crypto.verify.tx-check-valid-hit          | meter     | signature cache hits that occurred while validating transactions via checkValid
```
This was for consistency with the `crypto.verify.{hit,miss,total}` metrics. You're right it's unnecessary to report all three though. I removed the miss count.

bboston7 commented on src/overlay/test/OverlayTests.cpp:
```diff
@@ -3156,4 +3157,260 @@ TEST_CASE("background signature verification with missing account",
+
+        // Should be a single cache miss for the single signature
+        REQUIRE(finalMisses == 1);
+    }
```
I added one here and in the equivalent fee bump test

bboston7 commented:
Thanks for the feedback everyone! I believe I've addressed all the comments, so this is ready for another round of review. I've also rebased this on the current tip of `master`.

marta-lokhova reviewed (approved):
LGTM! One small nit, up to you if you want to address.

marta-lokhova commented:
Looks like the branch is out of date. Feel free to rebase and squash some intermediate commits (like reverts, etc) @bboston7, and we can merge then.

bboston7 commented:
I just rebased and squashed down to a few commits. I think this is ready to merge now!

bboston7 commented:
Really weird CI failure. Here's the error message:
```
2025-10-03T22:30:00.7807105Z local:9/49/100%/4.6s 2	error: invalid thread
2025-10-03T22:30:00.7807457Z 2	Process 249143 exited with status = 0 (0x00000000)
2025-10-03T22:30:00.7807964Z 2	Process 249143 launched: '/home/runner/work/stellar-core/stellar-core/src/stellar-core' (x86_64)
2025-10-03T22:30:00.7808586Z 2	(lldb) bt
```

The process exited with a successful exit code (`0`), and the last output from partition 2 prior to this failure was:
```
2025-10-03T22:30:00.4692650Z 2	..............................<done>
2025-10-03T22:30:00.4693071Z 2	
2025-10-03T22:30:00.4693844Z 2	[1;31m[0m[1;33m[0m[1;32m===============================================================================[0m
2025-10-03T22:30:00.4694353Z 2	
2025-10-03T22:30:00.4694849Z 2	[1;32mAll tests passed[0m (260391 assertions in 5 test cases)
2025-10-03T22:30:00.4695351Z 2	
2025-10-03T22:30:00.4695593Z 2	
2025-10-03T22:30:00.4695835Z 2	
```

So it looks like the tests in the partition did all pass, but for some reason the script thought they failed? I reran the tests from the partition locally and they all passed as well. I'm going to restart that job to see what happens on a rerun.
